### DISCUSSION TITLE
Can't get Solid to work in Point and Click Example

### DISCUSSION THREAD

**Discussion by jdiperla** [replit/kaboom#180] [Q&A]
This is the code and it seems to go right through the supposed solid object. No doubt I am doing something wrong. 

```
`kaboom();

// load mark sprite
loadMark();

const obj = add([
	sprite("mark"),
	area(),
	scale(1),
	solid(),
]);


// add to screen
const player = add([
	sprite("mark"),
	pos(),
	origin("center"),
	 
]);

let dest = vec2(0);

player.action(() => {
	obj.pushOutAll();
	player.moveTo(dest, 10);
});

mouseClick(() => {
	dest = mousePos();
});
`
```

**Comment by slmjkdbtl** [comment]
Try `pushOutAll` on the player? I agree the name is still a bit confusing here

**Comment by jdiperla** [comment]
If I do player.pushOutAll it doesnt work. If I do player.pushOutAll(); then it says it is not a function.

**Comment by slmjkdbtl** [comment]
oh because it's under `area()`, you'll have to give `area()` component to player

**Comment by jdiperla** [comment]
Yes! That was it. Now I can make point and click adventure games. Woo hoo. It pathfinds around the solid object nicely.

**Reply by slmjkdbtl** [reply]
Yees! Make it :D Come here with any question

**Reply by slmjkdbtl** [reply]
Oh no there's no really path finds yet
